node/src/service.rs

```diff
@@ -280,32 +278,26 @@ where
 		hwbench.clone(),
 	)
 	.await
-	.map_err(|e| match e {
-		RelayChainError::ServiceError(polkadot_service::Error::Sub(x)) => x,
-		s => s.to_string().into(),
-	})?;
-
-	let block_announce_validator =
-		BlockAnnounceValidator::new(relay_chain_interface.clone(), para_id);
+	.map_err(|e| sc_service::Error::Application(Box::new(e) as Box<_>))?;
```

Comment:
This was part of #171 and was mostly based on the refactor to the [parachain template v0.9.40 update](https://github.com/substrate-developer-hub/substrate-parachain-template/pull/153)

Reply:
What is the point in casting Box to Box? this is not needed in the parachain template either. It can be needed sometimes, but this doesn't seem to be the case here.